cmd/juju/cloud/show.go

```diff
@@ -130,6 +142,50 @@ func makeCloudDetails(cloud jujucloud.Cloud) *cloudDetails {
 		result.Regions = append(result.Regions, yaml.MapItem{r.Name, r})
 		result.RegionsMap[region.Name] = r
 	}
+	modelConfig := make(map[string]interface{})
+	for attr, value := range result.BootstrapConfig {
+		if !bootstrap.IsBootstrapAttribute(attr) {
+			modelConfig[attr] = value
+			delete(result.BootstrapConfig, attr)
+		}
+	}
+	if len(modelConfig) > 0 {
+		result.ModelConfig = modelConfig
+	}
+	// Look for config options specific to the cloud provider, if available.
+	provider, err := environs.Provider(cloud.Type)
+	if err != nil {
+		return result
+	}
+	// Not all providers have implemented the ProviderSchema interface,
+	// e.g. joyent, double check before using.
+	if m, ok := interface{}(provider).(environs.ProviderSchema); ok {
+		// providerSchema has all config options,
+		// including their descriptions and types.
+		providerSchema := m.Schema()
+		if providerSchema == nil {
+			return result
+		}
+		specifics := make(map[string]interface{})
+		// Check to make sure the ConfigSchemaSource
+		// interface was implemented by the provider.
+		if ps, ok := provider.(config.ConfigSchemaSource); ok {
+			// ps.ConfigSchema() returns the provider
+			// specific config option names, but no
+			// description etc.
+			for attr := range ps.ConfigSchema() {
+				if !providerSchema[attr].Secret {
```

Comment:
To avoid excessive indenting, we can just 
if providerSchema[attr].Secret {
    continue
}